scripts/flag_reviews.py

```diff
@@ -1,66 +1,69 @@
+            review_proximity = pd.Timedelta(timestamp - pd.to_datetime(duplicate_reviews_dict[dict_key]['updatedAt'])).seconds / 60
             if abs(review_proximity) < 300:
-                # print("With review proxmity ", review_proximity, " minutes")
-                if (safety_info == duplicate_reviews_dict[dict_key]['Safety Observations'].strip().lower()) and (useful_info == duplicate_reviews_dict[dict_key]['Useful Information'].strip().lower()):
-                    # print("duplicate found: ", safety_info, duplicate_reviews_dict[dict_key]['Safety Observations'].strip().lower(), "and useful info ", useful_info, duplicate_reviews_dict[dict_key]['Useful Information'].strip().lower())
-                    row['flag'] = 'Duplicate'
+                if (safety_info == duplicate_reviews_dict[dict_key]['safetyInfo'].strip().lower()) and (useful_info == duplicate_reviews_dict[dict_key]['availabilityInfo'].strip().lower() and pd.notnull(row['openingTime']) and pd.notnull(row['closingTime'])):
+                    # print("duplicate found: ", safety_info, duplicate_reviews_dict[dict_key]['safetyInfo'].strip().lower(), "and useful info ", useful_info, duplicate_reviews_dict[dict_key]['availabilityInfo'].strip().lower())
+                    df.at[index , 'flag'] = Flag.DUPLICATE
                     continue                    
                 # else: 
-                    # print("duplicate NOT found: ", safety_info, duplicate_reviews_dict[dict_key]['Safety Observations'].strip().lower(), "and useful info ", useful_info, duplicate_reviews_dict[dict_key]['Useful Information'].strip().lower())
+                    # print("duplicate NOT found: ", safety_info, duplicate_reviews_dict[dict_key]['safetyInfo'].strip().lower(), "and useful info ", useful_info, duplicate_reviews_dict[dict_key]['availabilityInfo'].strip().lower())
 
         else:
             duplicate_reviews_dict[dict_key] = row
 
-        if not row['Latitude'] or not row['Longitude']:
-            row['flag'] = 'Empty lat/long'
+        if not row['latitude'] or not row['longitude'] or np.isnan(row['latitude']) or np.isnan(row['longitude']):
+            df.at[index , 'flag'] = str(Flag.MISSING_LAT_LONG)
             continue
             
-        if store_name.isdigit() or (row['Address'] and store_name == row['Address'][:len(store_name)]):
-            row['flag'] = 'Invalid store name'
+        if store_name.isdigit() or (row['address'] and store_name == row['address'][:len(store_name)]):
+            df.at[index , 'flag'] = Flag.INVALID_STORE_NAME
             continue
 
         if len(useful_info) < 5 and len(safety_info) < 5:
-            row['flag'] = 'Very short reviews'
+            df.at[index , 'flag'] = Flag.VERY_SHORT_REVIEW
             continue
 
         if len(useful_info) < 10 and len(safety_info) < 10:
-            row['flag'] = 'Short reviews'
+            df.at[index , 'flag'] = Flag.SHORT_REVIEW
```

Comment:
this check for short review will overwrite previous flaggings. is that going to create problems? same for other flags

Reply:
ok, just realized you are `continue`ing after each condition. that will also imply we are only flagging once and ignore other flags.

Reply:
Yes that's right. It doesn't matter whether it's caught under the right category, I think what's more important is knowing whether it should be caught at all or not. If there's ever an explicit order, we can re-order the conditions accordingly.